### PR TITLE
Issue 718 refactor

### PR DESCRIPTION
**Pull request by captivus**
Resolves issue #718 and enhances documentation of each refactored module.

Tests running successfully on refactored codebase, with the exception of the misimplemented (missing) test for the AI module
image

### CODE REVIEW

File: gpt_engineer/cli/file_selector.py [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+import sys
+import tkinter as tk
+import tkinter.filedialog as fd
+
+from pathlib import Path
+from typing import List, Union
+
+from gpt_engineer.core.db import DB, DBs
+
+# ... [rest of the code] ...
```
Comment (UmerHA) [inline gpt_engineer/cli/file_selector.py:52]: ?

File: gpt_engineer/cli/main.py [hunk 2]
```diff
@@ -8,11 +35,11 @@
...
-from gpt_engineer.ai import AI
-from gpt_engineer.collect import collect_learnings
-from gpt_engineer.db import DB, DBs, archive
-from gpt_engineer.learning import collect_consent
-from gpt_engineer.steps import STEPS, Config as StepsConfig
+from gpt_engineer.core.ai import AI
+from gpt_engineer.cli.collect import collect_learnings
+from gpt_engineer.core.db import DB, DBs, archive
+from gpt_engineer.cli.learning import collect_consent
+from gpt_engineer.core.steps import STEPS, Config as StepsConfig
```
Comment (UmerHA) [inline gpt_engineer/cli/main.py:43]: i think we should order the imports

**Review by UmerHA — APPROVED** [review]
Awesome work! Have added minor comments

**Comment by UmerHA** [comment]
@captivus also, please fix the merge conflicts :)

File: gpt_engineer/cli/file_selector.py (hunk 1 above)
Comment (captivus) [reply gpt_engineer/cli/file_selector.py:52]: Fixed.

**Comment by UmerHA** [comment]
@captivus congrats on the merge - great work!

**Pull request merged**